tests/integ/test_session.py

```diff
@@ -331,6 +331,21 @@ def test_create_session_from_connection_with_noise_parameters(
         new_session.close()
 
 
+@pytest.mark.skipif(IS_IN_STORED_PROC, reason="Cannot create session in SP")
+def test_session_builder_app_name(session, db_parameters):
+    builder = session.builder
+    app_name = 'my_app'
+    expected_query_tag = f'APPNAME={app_name}'
+    same_session = builder.app_name(app_name).getOrCreate()
+    new_session = builder.app_name(app_name).configs(db_parameters).create()
```

Comment:
Would we want users to be able to add query tags and app name?

Reply:
I think the app_name is a specific kinds of query tag, if the user want to add additional query tags they should be able to do that. this app_name is only intent to set up before session creation. Do you know why the SessionBuilder getter always returns a new instance?